File: mobile/src/utils/format.ts
export function formatPrice(value: number | null | undefined, currency = "TRY"): string {
  if (value == null) return "-";
  const symbol = currency === "TRY" ? "₺" : currency === "USD" ? "$" : currency === "EUR" ? "€" : "";
  const formatted = Number(value).toLocaleString("tr-TR", {
    minimumFractionDigits: 0,
    maximumFractionDigits: 2,
  });
  return symbol ? `${formatted} ${symbol}` : `${formatted} ${currency}`;
}

export function formatPriceRange(
  min: number | null,
  max: number | null,
  currency = "TRY"
): string {
  if (min == null && max == null) return "Fiyat bilgisi yok";
  if (min != null && max != null && min !== max) {
    return `${formatPrice(min, currency)} - ${formatPrice(max, currency)}`;
  }
  return formatPrice(min ?? max, currency);
}

export function formatDistance(km: number | null | undefined): string | null {
  if (km == null) return null;
  if (km < 1) return `${Math.round(km * 1000)} m`;
  return `${km.toFixed(1)} km`;
}

const MONTHS_TR = [
  "Oca", "Şub", "Mar", "Nis", "May", "Haz",
  "Tem", "Ağu", "Eyl", "Eki", "Kas", "Ara",
];

export function formatDate(iso: string): string {
  // iso is "YYYY-MM-DD"
  const parts = iso.split("-");
  if (parts.length !== 3) return iso;
  const [y, m, d] = parts;
  const month = MONTHS_TR[Number(m) - 1] ?? m;
  return `${Number(d)} ${month} ${y}`;
}

export function todayIso(): string {
  const now = new Date();
  const y = now.getFullYear();
  const m = String(now.getMonth() + 1).padStart(2, "0");
  const d = String(now.getDate()).padStart(2, "0");
  return `${y}-${m}-${d}`;
}

export function addDaysIso(baseIso: string, days: number): string {
  const d = new Date(`${baseIso}T12:00:00`);
  d.setDate(d.getDate() + days);
  const y = d.getFullYear();
  const m = String(d.getMonth() + 1).padStart(2, "0");
  const day = String(d.getDate()).padStart(2, "0");
  return `${y}-${m}-${day}`;
}

File: android/app/src/main/java/com/rork/asistanrezervasyonandroid/data/Format.kt
package com.rork.asistanrezervasyonandroid.data

import java.text.SimpleDateFormat
import java.util.Calendar
import java.util.Date
import java.util.GregorianCalendar
import java.util.Locale
import java.util.TimeZone

/** Formatting helpers shared across cards — mirrors iOS Format enum. */
object Format {
    private val monthNames = listOf("Oca", "Şub", "Mar", "Nis", "May", "Haz", "Tem", "Ağu", "Eyl", "Eki", "Kas", "Ara")
    private val weekdayNames = listOf("Paz", "Pzt", "Sal", "Çar", "Per", "Cum", "Cmt")

    fun price(value: Double, currency: String = "TRY"): String {
        val symbol = if (currency == "TRY") "₺" else currency
        return if (value == Math.floor(value)) {
            "$symbol${value.toLong()}"
        } else {
            "$symbol${String.format(Locale.US, "%.2f", value)}"
        }
    }

    fun priceRange(min: Double?, max: Double?, currency: String): String? {
        if (min == null) return null
        if (max != null && max != min) {
            return "${price(min, currency)} – ${price(max, currency)}"
        }
        return price(min, currency)
    }

    fun distance(km: Double?): String? {
        if (km == null) return null
        if (km < 1) return "${(km * 1000).toInt()} m"
        return String.format(Locale.US, "%.1f km", km)
    }

    /** "2026-06-15" -> "15 Haz Pzt" */
    fun shortDate(iso: String): String {
        val parts = iso.split("-")
        if (parts.size != 3) return iso
        val y = parts[0].toIntOrNull() ?: return iso
        val m = parts[1].toIntOrNull() ?: return iso
        val d = parts[2].toIntOrNull() ?: return iso
        val cal = GregorianCalendar(y, m - 1, d)
        val weekday = cal.get(Calendar.DAY_OF_WEEK) - 1 // Calendar.SUNDAY == 1
        val month = if (m in 1..12) monthNames[m - 1] else ""
        return "$d $month ${weekdayNames[weekday]}"
    }

    /** Friendly label for next-available slot. */
    fun nextSlotLabel(slot: NextSlot, today: String): String {
        if (slot.date == today) return "Bugün ${slot.startTime}"
        return "${shortDate(slot.date)} • ${slot.startTime}"
    }

    val todayIso: String
        get() {
            val cal = Calendar.getInstance(TimeZone.getTimeZone("Europe/Istanbul"))
            return String.format(
                Locale.US, "%04d-%02d-%02d",
                cal.get(Calendar.YEAR), cal.get(Calendar.MONTH) + 1, cal.get(Calendar.DAY_OF_MONTH)
            )
        }

    /** Next 14 bookable dates in the business timezone. */
    fun dateOptions(count: Int = 14): List<String> {
        val cal = Calendar.getInstance(TimeZone.getTimeZone("Europe/Istanbul"))
        return (0 until count).map { offset ->
            val c = cal.clone() as Calendar
            c.add(Calendar.DAY_OF_MONTH, offset)
            String.format(
                Locale.US, "%04d-%02d-%02d",
                c.get(Calendar.YEAR), c.get(Calendar.MONTH) + 1, c.get(Calendar.DAY_OF_MONTH)
            )
        }
    }

    /** "09:00:00" -> "09:00" */
    fun hhmm(time: String): String = time.take(5)

    /** Relative time label for notifications ("5 dk önce"). */
    fun relativeTime(iso: String): String {
        val date = parseIso(iso) ?: return shortDate(iso.take(10))
        val interval = (System.currentTimeMillis() - date.time) / 1000
        return when {
            interval < 60 -> "Az önce"
            interval < 3600 -> "${interval / 60} dk önce"
            interval < 86400 -> "${interval / 3600} saat önce"
            interval < 604800 -> "${interval / 86400} gün önce"
            else -> shortDate(iso.take(10))
        }
    }

    private fun parseIso(iso: String): Date? {
        if (iso.length < 19) return null
        val df = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss", Locale.US)
        df.timeZone = TimeZone.getTimeZone("UTC")
        return runCatching { df.parse(iso.take(19)) }.getOrNull()
    }
}
